lib/broadcasts.ts

```diff
@@ -0,0 +1,73 @@
+import fs from "fs";
+import path from "path";
+import matter from "gray-matter";
+
+const BROADCAST_DATA_FOLDER = "schedule/broadcasts";
+
+export type Weekday = "Monday" | "Tuesday" | "Wednesday" | "Thursday" | "Friday" | "Saturday" | "Sunday";
+
+interface MatterData {
+  title: string;
+  subtitle: string;
+  weekday: Weekday;
+  startTime: string;
+  endTime: string;
+  isLive: boolean;
+}
+
+export type BroadcastData = MatterData & {
+  slug: string;
+  content: string;
+}
+
+export function getBroadcasts(): Array<BroadcastData> {
+  const broadcastsPath = fs.readdirSync(path.resolve(BROADCAST_DATA_FOLDER));
+
+  const broadcastsData = broadcastsPath.map((programPath) => {
+    return getMatterData(programPath);
+  });
+
+  return broadcastsData;
+}
+
+export function getBroadcastData(slug: string): BroadcastData {
```

Comment:
Some of these could be extracted to shared functions later.

Reply:
Sure let's get a form and will gather these in a common folder to avoid repeating ourselves :)